### PR TITLE
Make all email subjects configurable

### PR DESCRIPTION
**Pull request by raphaelm**

### CODE REVIEW

File: src/pretix/static/pretixcontrol/js/ui/mail.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ function preview_task_callback(data, jqXHR, status) {
         $('#' + data.item + '_panel').data('ajaxing', false);
         for (var m in data.msgs){
             var target = $('div[for=' + data.item + '][lang=' + m +']');
+            console.log(data.item, m, target);
```
Comment (wiffbi) [inline src/pretix/static/pretixcontrol/js/ui/mail.js:7]:
```suggestion
```

File: src/pretix/control/views/event.py [hunk 2]
```diff
@@ -732,9 +733,12 @@ def post(self, request, *args, **kwargs):
...
                     with language(self.supported_locale[idx], self.request.event.settings.region):
-                        msgs[self.supported_locale[idx]] = markdown_compile_email(
-                            v.format_map(self.placeholders(preview_item))
-                        )
+                        if 'subject' in k:
+                            msgs[self.supported_locale[idx]] = bleach.clean(v).format_map(self.placeholders(preview_item))
+                        else:
+                            msgs[self.supported_locale[idx]] = markdown_compile_email(
+                                v.format_map(self.placeholders(preview_item))
+                            )
```
Comment (wiffbi) [inline src/pretix/control/views/event.py:741]: Would it make sense to `bleach.clean` after paceholder-interpolation? And also make that `if` checking for mail_subject more specific? E.g.
```suggestion
                        v = v.format_map(self.placeholders(preview_item))
                        if k.startswith('mail_subject_'):
                            msgs[self.supported_locale[idx]] = bleach.clean(v)
                        else:
                            msgs[self.supported_locale[idx]] = markdown_compile_email(v)
```

File: src/pretix/control/views/organizer.py [hunk 3]
```diff
@@ -333,9 +334,12 @@ def post(self, request, *args, **kwargs):
...
                     with language(self.supported_locale[idx], self.request.organizer.settings.region):
-                        msgs[self.supported_locale[idx]] = markdown_compile_email(
-                            v.format_map(self.placeholders(preview_item))
-                        )
+                        if 'subject' in k:
+                            msgs[self.supported_locale[idx]] = bleach.clean(v).format_map(self.placeholders(preview_item))
+                        else:
+                            msgs[self.supported_locale[idx]] = markdown_compile_email(
+                                v.format_map(self.placeholders(preview_item))
+                            )
```
Comment (wiffbi) [inline src/pretix/control/views/organizer.py:342]: See above
```suggestion
                        v = v.format_map(self.placeholders(preview_item))
                        if k.startswith('mail_subject_'):
                            msgs[self.supported_locale[idx]] = bleach.clean(v)
                        else:
                            msgs[self.supported_locale[idx]] = markdown_compile_email(v)
```

**Review by wiffbi — COMMENTED** [review]
Looks good, just a minor comment regarding HTML-sanitizing on the email’s subject.

File: src/pretix/control/views/event.py (hunk 2 above)
Comment (raphaelm) [reply src/pretix/control/views/event.py:741]: > Would it make sense to bleach.clean after paceholder-interpolation?

Nope! The placeholders, in this case, will contain HTML, namely the snippet that makes them yellow and gives them an "I am a placeholder" tooltip.

> And also make that if checking for mail_subject more specific?

Yeah, maybe.

**Pull request merged**